doozerlib/runtime.py

```diff
@@ -644,56 +644,44 @@ def resolve_source(self, parent, meta):
 
         url = source_details["url"]
         branches = source_details['branch']
-        self.logger.info("Cloning source '%s' from %s as specified by group into: %s" % (alias, url, source_dir))
-        exectools.cmd_assert(
-            cmd=["git", "clone", url, source_dir],
-            retries=3,
-            on_retry=["rm", "-rf", source_dir],
-        )
+
         stage_branch = branches.get('stage', None)
         fallback_branch = branches.get("fallback", None)
         found = False
-        with Dir(source_dir):
-            if self.stage and stage_branch:
-                self.logger.info('Normal branch overridden by --stage option, using "{}"'.format(stage_branch))
-                branch = stage_branch
-            else:
-                branch = branches["target"]
-            self.logger.info("Attempting to checkout source '%s' branch %s in: %s" % (alias, branch, source_dir))
 
-            if branch != "master":
-                rc, out, err = exectools.cmd_gather(["git", "checkout", "-b", branch, "origin/%s" % branch])
-            else:
-                rc = 0
+        if self.stage and stage_branch:
+            self.logger.info('Normal branch overridden by --stage option, using "{}"'.format(stage_branch))
+            branch = stage_branch
+        else:
+            branch = branches["target"]
 
-            if rc == 0:
-                found = True
-            else:
+        base_clone = "git clone -b {} --single-branch {} --depth 1 {}"
+        clone_target = base_clone.format(branch, url, source_dir)
+        clone_fallback = base_clone.format(fallback_branch, url, source_dir)
+        retries = 3
+        while retries:
```

Comment:
could we get rid of the iterator and just do this instead:
```suggestion
        while _ in range(3):
```